backend/src/test/kotlin/org/loculus/backend/service/submission/UseNewerProcessingPipelineVersionTaskTest.kt

```diff
@@ -106,4 +108,58 @@ class UseNewerProcessingPipelineVersionTaskTest(
 
         assertThat(rowCount, `is`(rowCountAfterV2))
     }
+
+    @Test
+    fun `GIVEN multiple pipeline versions exist WHEN the version is bumped THEN old data is deleted`() {
+        // ... but only for that organism!
+
+        // create data for OTHER_ORGANISM - so we can check later that it doesn't get deleted
+        convenienceClient.prepareDefaultSequenceEntriesToApprovedForRelease(
+            organism = OTHER_ORGANISM,
+        )
+
+        val processedData = convenienceClient.submitDefaultFiles().submissionIdMappings.map {
+            PreparedProcessedData.successfullyProcessed(it.accession, it.version)
+        }
+
+        convenienceClient.extractUnprocessedData(pipelineVersion = 1)
+        convenienceClient.submitProcessedData(processedData, pipelineVersion = 1)
+        useNewerProcessingPipelineVersionTask.task()
+
+        convenienceClient.extractUnprocessedData(pipelineVersion = 2)
+        convenienceClient.submitProcessedData(processedData, pipelineVersion = 2)
+        useNewerProcessingPipelineVersionTask.task()
+
+        transaction {
+            assertThat(getExistingPipelineVersions(DEFAULT_ORGANISM), `is`(listOf(2L)))
+            assertThat(getExistingPipelineVersions(OTHER_ORGANISM), `is`(listOf(1L)))
```

Comment:
sorry I don't quite understand why  OTHER_ORGANISM is not being updated to a higher version with the `useNewerProcessingPipelineVersionTask.task()` statement

Reply:
Good question, this test comes from Felix.

My assumption is that the convenience client submits for the DEFAULT_ORGANISM, not other organism, hence the version is always still 1 for other organism. It won't be bumped as there's nothing to be bumped for other organism? Makes sense?

Reply:
True the default organism is DEFAULT_ORGANISM - maybe we could add a comment?